msal/src/main/java/com/microsoft/identity/client/internal/CommandParametersAdapter.java

```diff
@@ -757,6 +767,192 @@ public static SignInSubmitPasswordCommandParameters createSignInSubmitPasswordCo
+     * @throws ClientException
+     */
+    public static MFAChallengeCommandParameters createMFADefaultChallengeCommandParameters(
+            @NonNull final NativeAuthPublicClientApplicationConfiguration configuration,
+            @NonNull final OAuth2TokenCache tokenCache,
+            @NonNull final String continuationToken,
+            @NonNull final String correlationId,
+            final List<String> scopes) throws ClientException {
+
+        final NativeAuthCIAMAuthority authority = ((NativeAuthCIAMAuthority) configuration.getDefaultAuthority());
+
+        final AbstractAuthenticationScheme authenticationScheme = AuthenticationSchemeFactory.createScheme(
+                AndroidPlatformComponentsFactory.createFromContext(configuration.getAppContext()),
+                null
+        );
+
+        final MFAChallengeCommandParameters commandParameters =
+                MFAChallengeCommandParameters.builder()
+                        .platformComponents(AndroidPlatformComponentsFactory.createFromContext(configuration.getAppContext()))
+                        .applicationName(configuration.getAppContext().getPackageName())
+                        .applicationVersion(getPackageVersion(configuration.getAppContext()))
+                        .clientId(configuration.getClientId())
+                        .isSharedDevice(configuration.getIsSharedDevice())
+                        .redirectUri(configuration.getRedirectUri())
+                        .oAuth2TokenCache(tokenCache)
+                        .requiredBrokerProtocolVersion(configuration.getRequiredBrokerProtocolVersion())
+                        .sdkType(SdkType.MSAL)
+                        .sdkVersion(PublicClientApplication.getSdkVersion())
+                        .powerOptCheckEnabled(configuration.isPowerOptCheckForEnabled())
+                        .authority(authority)
+                        .authenticationScheme(authenticationScheme)
+                        .continuationToken(continuationToken)
+                        .scopes(scopes)
+                        .challengeType(configuration.getChallengeTypes())
+                        .correlationId(correlationId)
+                        .build();
+
+        return commandParameters;
+    }
+
+    /**
+     * Creates command parameter for [{@link com.microsoft.identity.common.nativeauth.internal.commands.MFAChallengeCommand}] of Native Auth
+     * @param configuration PCA configuration
+     * @param tokenCache token cache for storing results
+     * @param correlationId correlation ID to use in the API request, taken from the previous request in the flow
+     * @param continuationToken continuation token
+     * @param authMethod the user's authentication method that is used to perform the challenge operation
+     * @return Command parameter object
+     * @throws ClientException
+     */
+    public static MFASelectedChallengeCommandParameters createMFASelectedChallengeCommandParameters(
```

Comment:
Might make more sense to have one common method `createMFASelectedChallengeCommandParameters` that takes in all the needed parameters, and just have other methods expect less parameters but pass null for other fields they don't pass. That way there's less code to maintain.